apis/core/lib/domain/table/csvw.ts

```diff
@@ -0,0 +1,39 @@
+import * as Csvw from '@rdfine/csvw'
+import { TableMixin as CsvwTable } from '@rdfine/csvw'
+import { CsvSourceMixin, TableMixin } from '@cube-creator/model'
+import cf from 'clownface'
+import $rdf from 'rdf-ext'
+import { csvw } from '@tpluscode/rdf-ns-builders'
+import { NamedNode } from 'rdf-js'
+import { ResourceStore } from '../../ResourceStore'
+import { resourceStore } from '../resources'
+
+interface Command {
+  tableResource: NamedNode
+  resources?: ResourceStore
+}
+
+export async function createCsvw({
+  tableResource,
+  resources = resourceStore(),
+}: Command): Promise<Csvw.Table> {
+  const table = new TableMixin.Class(await resources!.get(tableResource))
+  const source = new CsvSourceMixin.Class(await resources!.get(table.csvSource.id))
+
+  const {
+    dialect: {
+      quoteChar, delimiter, encoding, doubleQuote,
+    },
+  } = source
+
+  return new CsvwTable.Class(cf({ dataset: $rdf.dataset(), term: table.csvw }), {
+    url: source.associatedMedia.contentUrl?.value,
+    dialect: {
+      types: [csvw.Dialect],
+      quoteChar,
+      delimiter,
+      encoding,
+      doubleQuote,
+    },
+  }) as any
+}
```

Comment:
By annotating JS classes it is possible to avoid clownface pointers and directly using RDF vocabularies.

This also provides full type checking